packages/web/pages/stake/index.tsx

```diff
@@ -71,33 +85,94 @@ export const Staking: React.FC = observer(() => {
     return { currency: osmo, amount: primitiveAmount.amount, denom: osmo };
   }, [osmo, primitiveAmount]);
 
+  const delegationQuery = cosmosQueries.queryDelegations.getQueryBech32Address(
+    account?.address ?? ""
+  );
+
+  const userValidatorDelegations = delegationQuery.delegations;
+
+  const usersValidatorsMap = useMemo(() => {
+    const delegationsMap = new Map<string, StakingType.Delegation>();
+
+    userValidatorDelegations.forEach((delegation: StakingType.Delegation) => {
+      delegationsMap.set(delegation.delegation.validator_address, delegation);
+    });
+
+    return delegationsMap;
+  }, [userValidatorDelegations]);
+
+  const [amountDefault, setAmountDefault] = useState<AmountDefault>("input");
```

Comment:
nit: can you move this up with the other useState hooks?

Reply:
I think it can be removed. Amount config, where the half and max are implemented, has a `fraction` member. Instead of storing state separately in a useState, we can use `fraction === 0.5` for half and `fraction ==== 1` for max.

Reply:
Oh wait no, I'm guessing the data team asked to track whether the amount was input via the 'half' button, the 'max' button, or manually 'input', in which case we need to track that